debian/postinst

```diff
@@ -142,23 +141,50 @@ case "$1" in
             chmod 770 /var/lib/jigasi
         fi
 
+        JIGASI_XMPPPWD=$(echo -n $JIGASI_XMPP_SECRET | base64 -)
+
         # populate the config with debconf values
-        sed -i "s/<<JIGASI_SIPUSER>>/$JIGASI_SIPUSER/g" /etc/jitsi/jigasi/sip-communicator.properties
-        sed -i "s/<<JIGASI_SIPPWD>>/$JIGASI_SIPPWD/g" /etc/jitsi/jigasi/sip-communicator.properties
+        JIGASI_CONFIG="/etc/jitsi/jigasi/sip-communicator.properties"
+        sed -i "s/<<JIGASI_SIPUSER>>/$JIGASI_SIPUSER/g" $JIGASI_CONFIG
+        sed -i "s/<<JIGASI_SIPPWD>>/$JIGASI_SIPPWD/g" $JIGASI_CONFIG
         SIPSERVER=`echo $JIGASI_SIPUSER | cut -d'@' -f 2`
-        sed -i "s/<<JIGASI_SIPSERVER>>/$SIPSERVER/g" /etc/jitsi/jigasi/sip-communicator.properties
-        sed -i "s/<<DOMAIN_BASE>>/$JVB_HOSTNAME/g" /etc/jitsi/jigasi/sip-communicator.properties
+        sed -i "s/<<JIGASI_SIPSERVER>>/$SIPSERVER/g" $JIGASI_CONFIG
+        sed -i "s/<<DOMAIN_BASE>>/$JVB_HOSTNAME/g" $JIGASI_CONFIG
+        sed -i "s/<<JIGASI_XMPPPWD>>/$JIGASI_XMPPPWD/g" $JIGASI_CONFIG
 
         PROSODY_HOST_CONFIG="/etc/prosody/conf.avail/$JVB_HOSTNAME.cfg.lua"
         if [ -f $PROSODY_HOST_CONFIG ]; then
-            JIGASI_PROSODY="Component \"callcontrol.$JVB_HOSTNAME\" component_secret = \"$JIGASI_SECRET\""
-            # update prosody config
-            if [ ! "$(grep "Component \"callcontrol.$JVB_HOSTNAME\"" $PROSODY_HOST_CONFIG)" ]; then
-                echo $JIGASI_PROSODY >> $PROSODY_HOST_CONFIG
-                invoke-rc.d prosody restart
+            USER_EXISTS_CHECK=`prosodyctl adduser jigasi@auth.$JVB_HOSTNAME < /dev/null || true`
+            if [ ! "$USER_EXISTS_CHECK" = "That user already exists" ]; then
+                prosodyctl register jigasi auth.$JVB_HOSTNAME $JIGASI_XMPP_SECRET || true
+            fi
+        fi
+
+        JICOFO_CONFIG="/etc/jitsi/jicofo/jicofo.conf"
+        if [ -f $JICOFO_CONFIG ]; then
+            # add brewery room to jicofo config if it is not there
+            RES=0
+            hocon -f $JICOFO_CONFIG get jicofo.jigasi.brewery > /dev/null 2>&1 || RES=$? || true
```

Comment:
Is this equivalent?
```
if ! hocon -f $JICOFO_CONFIG get jicofo.jigasi.brewery > /dev/null 2>&1 ;then
    hocon -f $JICOFO_CONFIG set jicofo.jigasi.brewery \"JigasiBrewery@internal.auth.meet.damencho.net\"
fi
```

Reply:
Nope that doesn't work, I tested that first but it gives an error and the script is stopped

Reply:
I don't understand why. This runs as expected for me:
```bash
$ cat test.sh
#!/bin/bash
set -e

echo "" > test.conf

if ! hocon -f test.conf get nonexistent.key > /dev/null 2>&1 ; then
	echo "The key is missing"
fi

echo "This still prints"
```

```
$ ./test.sh
The key is missing
This still prints
```

Reply:
Cause it is not bash that runs it. Ok let me change it in a commit so I will build the debian package and test it and will show you.

Reply:
Ok, it works. I'm testing this the 5th time already not sure what was the problem before :(
but the post inst script was failing through the dpkg and was passing when I was executing it in bash console